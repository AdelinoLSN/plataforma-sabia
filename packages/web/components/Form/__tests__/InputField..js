import React from 'react';
import { render, fireEvent } from 'test-utils';

import { waitFor } from '@testing-library/react';
import { Form, InputField } from '..';

describe('InputField', () => {
	test('renders properly and on submit shows validation errors', async () => {
		const onSubmit = jest.fn(() => {});
		const { container, getByLabelText, findByText } = render(
			<Form onSubmit={onSubmit}>
				<InputField
					name="test-input"
					label="Test Input"
					type="text"
					help={<p>Help Text</p>}
					validation={{ required: true }}
				/>
			</Form>,
		);

		const input = getByLabelText(/test input/i);
		expect(input).toHaveAttribute('type', 'text');

		fireEvent.submit(container.querySelector('form'));
		await findByText('error:requiredField');
		expect(onSubmit).not.toHaveBeenCalled();
		expect(container).toMatchSnapshot();
	});

	test('does not trigger validation if not required', async () => {
		const onSubmit = jest.fn(() => {});
		const { container, getByLabelText } = render(
			<Form onSubmit={onSubmit}>
				<InputField
					name="test-input"
					label="Test Input"
					type="text"
					validation={{ required: false }}
				/>
			</Form>,
		);

		const input = getByLabelText(/test input/i);
		expect(input).toHaveAttribute('type', 'text');

		fireEvent.submit(container.querySelector('form'));
		await waitFor(() => expect(onSubmit).toHaveBeenCalled());

		expect(container).toMatchSnapshot();
	});
});
