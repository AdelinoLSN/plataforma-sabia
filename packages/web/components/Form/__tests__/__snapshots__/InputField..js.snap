// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField does not trigger validation if not required 1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 14px;
}

.c1 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 600;
}

.c5 {
  color: hsla(38,100%,51%);
  margin: 0 0 1rem 0;
  display: inline-block;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.2rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="test-input"
      >
        Test Input
      </label>
      <div
        class="c3"
      >
        <input
          aria-label="Test Input"
          aria-required="false"
          class="c4"
          id="test-input"
          name="test-input"
          placeholder=""
          type="text"
        />
      </div>
      <span
        class="c5"
      />
    </div>
  </form>
</div>
`;

exports[`InputField renders properly and on submit shows validation errors 1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 14px;
}

.c1 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 > input {
  border: 1px solid hsla(38,100%,51%);
}

.c1 > .react-select-container .react-select__control {
  border-color: hsla(38,100%,51%);
}

.c2 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 600;
}

.c6 {
  color: hsla(38,100%,51%);
  margin: 0 0 1rem 0;
  display: inline-block;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  width: 4rem;
  height: 4rem;
  margin: 0.6rem 0.5rem;
  fill: hsla(0,0%,80%);
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c5:hover {
  cursor: pointer;
  fill: hsla(0,0%,7%);
}

.c4 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.2rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="test-input"
      >
        Test Input
      </label>
      <div
        class="c3"
      >
        <input
          aria-label="Test Input"
          aria-required="true"
          class="c4"
          id="test-input"
          name="test-input"
          placeholder=""
          type="text"
        />
        <svg
          class="c5"
          currentItem="false"
          data-for="test-input"
          data-tip="true"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 708c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm62.9-219.5a48.3 48.3 0 0 0-30.9 44.8V620c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-21.5c0-23.1 6.7-45.9 19.9-64.9 12.9-18.6 30.9-32.8 52.1-40.9 34-13.1 56-41.6 56-72.7 0-44.1-43.1-80-96-80s-96 35.9-96 80v7.6c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V420c0-39.3 17.2-76 48.4-103.3C430.4 290.4 470 276 512 276s81.6 14.5 111.6 40.7C654.8 344 672 380.7 672 420c0 57.8-38.1 109.8-97.1 132.5z"
          />
        </svg>
        <div
          class="__react_component_tooltip uuid place-right type-dark"
          data-id="tooltip"
          id="test-input"
        >
          <style>
            
  	.uuid {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.uuid.place-top {
        margin-top: -10px;
    }
    .uuid.place-top::before {
        border-top: 8px solid transparent;
    }
    .uuid.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .uuid.place-bottom {
        margin-top: 10px;
    }
    .uuid.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .uuid.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .uuid.place-left {
        margin-left: -10px;
    }
    .uuid.place-left::before {
        border-left: 8px solid transparent;
    }
    .uuid.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .uuid.place-right {
        margin-left: 10px;
    }
    .uuid.place-right::before {
        border-right: 8px solid transparent;
    }
    .uuid.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  
          </style>
          <p>
            Help Text
          </p>
        </div>
      </div>
      <span
        class="c6"
      >
        error:requiredField
      </span>
    </div>
  </form>
</div>
`;
